lua/apisix/discovery/eureka.lua

```diff
@@ -0,0 +1,178 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local local_conf         = require("apisix.core.config_local").local_conf()
+local http               = require("resty.http")
+local core               = require("apisix.core")
+local ipairs             = ipairs
+local ngx_timer_at       = ngx.timer.at
+local ngx_timer_every    = ngx.timer.every
+local string_sub         = string.sub
+local string_find        = string.find
+local log                = core.log
+
+local applications
+local useragent = 'ngx_lua-apisix/v' .. core.version.VERSION
+
+local _M = {
+    version = 1.0,
+}
+
+
+local function service_info()
+    if not local_conf.eureka or not local_conf.eureka.urls then
+        log.error("do not set eureka.urls")
+        return
+    end
+
+    local urls = local_conf.eureka.urls
+    local basic_auth
+    local url = urls[math.random(#urls)]
```

Comment:
Random selection is not good.

Add health check to get healthy nodes.

We can add todo comments first.

Reply:
have add todo